File: app/src/main/java/br/com/vicentec12/mygames/ui/console/ConsoleAdapter.java
package br.com.vicentec12.mygames.ui.console;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ListAdapter;

import br.com.vicentec12.mygames.data.model.ConsoleWithGames;
import br.com.vicentec12.mygames.databinding.ItemConsoleBinding;
import br.com.vicentec12.mygames.interfaces.OnItemClickListener;

public class ConsoleAdapter extends ListAdapter<ConsoleWithGames, ConsoleHolder> {

    private OnItemClickListener mOnItemClickListener;

    protected ConsoleAdapter() {
        super(ConsoleWithGames.getDiffItemCallback());
    }

    public void setOnItemClickListener(OnItemClickListener mOnItemClickListener) {
        this.mOnItemClickListener = mOnItemClickListener;
    }

    @NonNull
    @Override
    public ConsoleHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemConsoleBinding mBinding = ItemConsoleBinding.inflate(LayoutInflater.from(parent.getContext()),
                parent, false);
        return new ConsoleHolder(mBinding, mOnItemClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull ConsoleHolder holder, int position) {
        holder.bind(getItem(position));
    }

}

File: app/src/main/java/br/com/vicentec12/mygames/ui/console/ConsoleAdapter.kt
package br.com.vicentec12.mygames.ui.console

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import br.com.vicentec12.mygames.data.model.ConsoleWithGames
import br.com.vicentec12.mygames.databinding.ItemConsoleBinding
import br.com.vicentec12.mygames.interfaces.OnItemClickListener

class ConsoleAdapter(
        val mOnItemClickListener: OnItemClickListener?
) : ListAdapter<ConsoleWithGames, ConsoleHolder>(ConsoleWithGames.DIFF_UTIL_CALLBACK) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ConsoleHolder(ItemConsoleBinding
            .inflate(LayoutInflater.from(parent.context), parent, false), mOnItemClickListener)

    override fun onBindViewHolder(holder: ConsoleHolder, position: Int) {
        holder.bind(getItem(position))
    }

}